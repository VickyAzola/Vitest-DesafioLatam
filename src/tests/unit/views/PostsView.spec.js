import { describe, it, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'
import PostsView from '@/views/PostsView.vue'

describe('PostsView', () => { 
    
    const wrapper = mount(PostsView)

    it('El HTML del componente se mantiene durante las pruebas', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
    
    test('Probando la existencia de la vista PostsView ', async () => {
        const router = createRouter({
            history: createWebHistory(),
            routes: [{
                path: '/posts',
                name: 'PostsViewVue',
                component: PostsView
            }],
        })

        router.push('/posts')
        await router.isReady()

        const wrapper = mount(PostsView, {
            global: {
                plugins: [router]
            }
        })

        expect(wrapper.findComponent(PostsView).exists()).toBe(true)
    })  

    test('Tiene una lista de posts, con el texto "post.name"', async () => {
        wrapper.vm.posts = [
            { id: 1, name: 'Post 1' },
            { id: 2, name: 'Post 2' },
            { id: 3, name: 'Post 3' },
            { id: 4, name: 'Post 4' }
        ]

        await wrapper.vm.$nextTick()

        const li = wrapper.findAll('li')

        expect(li[0].text()).toBe('Post 1')
        expect(li[3].text()).toBe('Post 4')
        expect(li).toHaveLength(4)
    })
})