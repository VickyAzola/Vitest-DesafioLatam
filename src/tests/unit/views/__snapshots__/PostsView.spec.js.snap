// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostsView > El HTML del componente se mantiene durante las pruebas 1`] = `
"<div data-v-148b760b="" class="posts">
  <h1 data-v-148b760b="">Vista de Posts</h1>
  <ul data-v-148b760b="">
    <li data-v-148b760b="">Post 1</li>
    <li data-v-148b760b="">Post 2</li>
    <li data-v-148b760b="">Post 3</li>
    <li data-v-148b760b="">Post 4</li>
  </ul>
</div>"
`;
